av1an-core/src/settings.rs

```diff
@@ -574,24 +583,30 @@ properly into a mkv file. Specify mkvmerge as the concatenation method by settin
         self.sc_pix_format,
         self.sc_method,
         self.sc_downscale_height,
-      ),
-      SplitMethod::None => Ok(Vec::new()),
-    }
+      )?,
+      SplitMethod::None => (Vec::new(), self.input.frames()),
+    })
   }
 
   // If we are not resuming, then do scene detection. Otherwise: get scenes from
   // scenes.json and return that.
   fn split_routine(&mut self) -> anyhow::Result<Vec<usize>> {
     let scene_file = self
       .scenes
-      .clone()
-      .unwrap_or_else(|| Path::new(&self.temp).join("scenes.json"));
+      .as_ref()
+      .map(|path| Cow::Borrowed(path.as_path()))
+      .unwrap_or_else(|| Cow::Owned(Path::new(&self.temp).join("scenes.json")));
 
-    let mut scenes = if (self.scenes.is_some() && scene_file.exists()) || self.resume {
-      crate::split::read_scenes_from_file(scene_file.as_path())?.0
+    let scenes = if (self.scenes.is_some() && scene_file.exists()) || self.resume {
+      crate::split::read_scenes_from_file(scene_file.as_ref())?
     } else {
       self.calc_split_locations()?
     };
+    self.frames = scenes.1;
+    get_done()
+      .frames
+      .store(self.frames, atomic::Ordering::SeqCst);
+    let mut scenes = scenes.0;
```

Comment:
Huh, that is what I tried to do originally but I got a syntax error from the destructuring assignment, so I assumed it only worked on tuple literals. Turns out it was actually because I was trying to do `let mut (scenes, frames)` which is incorrect, and the `mut` applies to each individual item (i.e. `let (mut scenes, frames) = ...`). Thanks, I will update it.